socks5.py

```diff
@@ -267,10 +344,36 @@ def handle_connect(self, address, port):
         except Exception:
             pass
 
-        logging.info('%s: shutting down', log_tag)
+        logging.debug('%s: shutting down', log_tag)
         self.server.close_request(self.request)
 
+
+    def connect_to_address(self, address, port, family, log_tag, timeout=None):
+        try:
+            remote = socket.socket(family, socket.SOCK_STREAM)
+            remote.settimeout(timeout)
+            if CONNECT_HOST:
```

Comment:
Won't CONNECT_HOST (and CONNECT_HOST["ipv4"], CONNECT_HOST["ipv6"]) always be defined? If so there's no need for redundant checks here.

Reply:
Will fix in the next commit.